tests/Makefile

```
@@ -8,6 +8,15 @@ check: clean client_test server_test message_formatting_tests run
 build: clean client_test server_test message_formatting_tests
 
 LINKING_FLAGS = -lpthread  #-fsanitize=address
+ifneq (,$(wildcard ../libutil.so))
+	LINKING_FLAGS += -L.. -lutil
+endif
+ifneq (,$(wildcard ../target/release/libutil.so))
+	LINKING_FLAGS += -L../target/release -lutil
+endif
+ifneq (,$(wildcard ../target/debug/libutil.so))
+	LINKING_FLAGS += -L../target/debug -lutil
```

If you also extend the RPATH than the LD_LIBRARY_PATH becomes unnecessary.

```suggestion
 	LINKING_FLAGS += -L../target/debug -lutil -Wl,-rpath,../target/debug
```

I am currently on holiday, otherwise I would do the change myself.

---

Instead of the `ROOT_DIR` definition, I would rather use `$(CURDIR)/..`
e.g.:
```
LINKING_FLAGS += -L../target/debug -lutil -Wl,-rpath,$(CURDIR)/../target/debug
```

---

sure.